code/game/machinery/telecomms/computers/logbrowser.dm

```diff
@@ -1,212 +1,158 @@
-
-			if("release")
-				servers = list()
-				screen = 0
-
-			if("mainmenu")
-				screen = 0
-
-			if("scan")
-				if(servers.len > 0)
-					temp = "<font color = #D70B00>- FAILED: CANNOT PROBE WHEN BUFFER FULL -</font color>"
-
+		if(MAIN_VIEW)
+			// If there are servers in the buffer, send them
+			var/list/found_servers = list()
+			for (var/obj/machinery/telecomms/server/server in servers)
+				found_servers += list(list("ref"=REF(server), "name"=server.name, "id"=server.id))
+			data["servers"] = found_servers
+		if(SERVER_VIEW)
+			// Send selected server data
+			var/list/server_out = list()
+			server_out["name"] = SelectedServer.name
+			server_out["traffic"] = SelectedServer.totaltraffic
+			// Get the messages on this server
+			var/list/packets = list()
+			for(var/datum/comm_log_entry/packet in SelectedServer.log_entries)
+				var/list/packet_out = list()
+				
+				packet_out["name"] = packet.name
+				
+				var/datum/language/language = packet.parameters["language"]
+				var/datum/language/language_instance = GLOB.language_datum_instances[language]
+				packet_out["language"] = language_instance.name
+				var/message_out = ""
+				var/message_in = packet.parameters["message"]
+				if(universal_translate || user.has_language(language))
+					// Known language or translator, show plain text
+					message_out = "\"[message_in]\""
+				else if(!user.has_language(language))
+					// Language unknown: scramble
+					message_out = "\"[language_instance.scramble(message_in)]\""
 				else
-					for(var/obj/machinery/telecomms/server/T in urange(25, src))
-						if(T.network == network)
-							servers.Add(T)
-
-					if(!servers.len)
-						temp = "<font color = #D70B00>- FAILED: UNABLE TO LOCATE SERVERS IN \[[network]\] -</font color>"
-					else
-						temp = "<font color = #336699>- [servers.len] SERVERS PROBED & BUFFERED -</font color>"
-
-					screen = 0
-
-	if(href_list["delete"])
-
-		if(!src.allowed(usr) && !(obj_flags & EMAGGED))
-			to_chat(usr, span_danger("ACCESS DENIED."))
-			return
-
-		if(SelectedServer)
-
-			var/datum/comm_log_entry/D = SelectedServer.log_entries[text2num(href_list["delete"])]
-
-			temp = "<font color = #336699>- DELETED ENTRY: [D.name] -</font color>"
-
-			SelectedServer.log_entries.Remove(D)
-			qdel(D)
-
-		else
-			temp = "<font color = #D70B00>- FAILED: NO SELECTED MACHINE -</font color>"
-
-	if(href_list["network"])
-
-		var/newnet = stripped_input(usr, "Which network do you want to view?", "Comm Monitor", network)
-
-		if(newnet && ((usr in range(1, src)) || issilicon(usr)))
-			if(length(newnet) > 15)
-				temp = "<font color = #D70B00>- FAILED: NETWORK TAG STRING TOO LENGHTLY -</font color>"
-
-			else
-
-				network = newnet
-				screen = 0
-				servers = list()
-				temp = "<font color = #336699>- NEW NETWORK TAG SET IN ADDRESS \[[network]\] -</font color>"
-
-	updateUsrDialog()
-	return
-
-/obj/machinery/computer/telecomms/server/attackby()
+					message_out = "(Uninteligible)"
```

Comment:
```suggestion
					message_out = "(Unintelligible)"
```

Reply:
clearly i am the unintelligible one here :pensive: